pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -57,8 +58,11 @@ func (r *Revision) Validate(ctx context.Context) *apis.FieldError {
 
 // Validate ensures RevisionTemplateSpec is properly configured.
 func (rt *RevisionTemplateSpec) Validate(ctx context.Context) *apis.FieldError {
+	allowZeroInitialScale := false
+	autoscalerConfig := apisconfig.FromContextOrDefaults(ctx).Autoscaler
+	allowZeroInitialScale = autoscalerConfig.AllowZeroInitialScale
```

Comment:
```suggestion
	allowZeroInitialScale = apisconfig.FromContextOrDefaults(ctx).Autoscaler.AllowZeroInitialScale
```

For all other occasions too.